File: src/main/java/ipp/ValidateJob.java
package ipp;

import static de.gmuth.ipp.client.IppTemplateAttributes.documentFormat;
import static de.gmuth.ipp.client.IppTemplateAttributes.printerResolution;
import static de.gmuth.ipp.core.IppResolution.Unit.DPI;

import de.gmuth.ipp.client.IppColorMode;
import de.gmuth.ipp.client.IppExchangeException;
import de.gmuth.ipp.client.IppMedia;
import de.gmuth.ipp.client.IppPrinter;
import de.gmuth.ipp.client.IppSides;
import de.gmuth.ipp.core.IppAttribute;
import de.gmuth.ipp.core.IppResponse;

public class ValidateJob {

  public static void main(String[] args) {
    IppPrinter ippPrinter = new IppPrinter("ipp://localhost:8632/printers/laser");

    IppResponse ippValidationResponse;
    try {
      ippValidationResponse = ippPrinter.validateJob(
        documentFormat("application/pdf"),
        printerResolution(600, DPI),
        IppColorMode.Color,
        IppSides.TwoSidedLongEdge,
        new IppMedia.Collection(null, null, "manual", null)
      );

    } catch (IppExchangeException ippExchangeException) {
      ippExchangeException.logDetails();
      System.out.println(ippExchangeException);
      ippValidationResponse = ippExchangeException.getIppResponse();
    }

    System.out.println("status: " + ippValidationResponse.getStatus());
    if (ippValidationResponse.getUnsupportedGroup().size() > 0) {
      System.out.println("unsupported attributes or values:");
      for (IppAttribute ippAttribute : ippValidationResponse.getUnsupportedGroup().values()) {
        System.out.println(ippAttribute);
      }
    }
  }

}

File: src/main/kotlin/ipp/ValidateJob.kt
package ipp

import de.gmuth.ipp.client.*
import de.gmuth.ipp.client.IppTemplateAttributes.documentFormat
import de.gmuth.ipp.client.IppTemplateAttributes.printerResolution
import de.gmuth.ipp.core.IppResolution.Unit.DPI
import de.gmuth.log.Logging
import ipp.Printers.laser

fun main() {
    val log = Logging.getLogger {}
    val ippPrinter = IppPrinter(laser)

    val ippValidationResponse = try {
        ippPrinter.validateJob(
            documentFormat("application/pdf"),
            printerResolution(600, DPI),
            IppColorMode.Color,
            IppSides.TwoSidedLongEdge,
            IppMedia.Collection(source = "manual")
        )

    } catch (ippExchangeException: IppExchangeException) {
        //log.error(ippExchangeException) {"failed to validate job"}
        log.logWithCauseMessages(ippExchangeException)
        ippExchangeException.response
    }

    ippValidationResponse?.run {
        if (unsupportedGroup.size > 0) {
            for (ippAttribute in unsupportedGroup.values) {
                log.info { ippAttribute }
            }
        }
    }

//    status: client-error-attributes-or-values-not-supported
//    unsupported attributes or values:
//    media-col (collection) = {media-source=manual}

}